timer/config.go

```diff
@@ -0,0 +1,64 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package timer
+
+import (
+	"time"
+)
+
+// UnlimitedBudget is a special value for the budget that means the timer has no budget
+var UnlimitedBudget = ^time.Duration(0) - 1
```

Comment:
This is a pretty convoluted way to write `time.Duration(-2)`. And then why not `-1`?

I think it'd be best to use a more extreme value, as that'd stand out more when debugging. Since it's an `int64`, maybe the max `int64` as an approximation of "infinity":

```suggestion
var UnlimitedBudget = time.Duration(math.MaxInt64)
```

Reply:
Ok so goland was broken I thought this value did not exist